### ISSUE TITLE
[MaterialTextView] Add possibility to set drawable size

### ISSUE THREAD

**Issue by marynovskyi** [material-components/material-components-android#1439]
Hi,

It would be great if we had a possibility to set custom drawable start/end/top/bottom size, since currently it's pretty useless attributes, cause we can't control them except for space between drawable and text. If we could control more parameters when set a drawable to textView it would be much easier to use and really helpful

**Comment by consp1racy** [comment]
`MaterialButton` already has the following attributes:

* `app:icon`
* `app:iconSize`
* `app:iconGravity`
* `app:iconPadding`

Do you need anything else?

**Comment by marynovskyi** [comment]
Hi @consp1racy ,

I've got it - most cases when we add icon to textView is when that textView is used for click, so we can use button for it.
Thanks, I'm closing this issue

**Comment by MuntashirAkon** [comment]
For those who are here for this feature, a better option is to use [ImageSpan](https://developer.android.com/reference/android/text/style/ImageSpan).

**Comment by mecoFarid** [comment]
@marynovskyi You asked for `MaterialTextView` they suggested totally unrelated component and you said "yeah, that is right"?

**Comment by OhhhThatVarun** [comment]
@consp1racy I tried using `MaterialButton` in the `RecyclerView` instead of `TextView` and the performance drop is crazy. I checked the SystemTrace and a single `MaterialButton` is taking around 4ms to inflate I had 3 of them inside a `MaterialButtonToggleGroup`. Use-case I had was I just wanted to show the no of views, comment and share in my `ViewHolder`. So, having `MaterialTextView` with resizable drawable would absolutely help.
